### PR TITLE
[X86] Support DomainReassignment for APX NDD instructions

### PR DESCRIPTION
**Pull request by FreddyLeaf**

### CODE REVIEW

File: llvm/test/CodeGen/X86/domain-reassignment.mir
Comment (KanRobert) [inline llvm/test/CodeGen/X86/domain-reassignment.mir:1]: You should not remove tests for non-NDD variant.

Comment (FreddyLeaf) [reply llvm/test/CodeGen/X86/domain-reassignment.mir:1]: Addressed.

**Comment by KanRobert** [comment]
Do not rename the test name after copy and put it in the apx directory.

**Comment by FreddyLeaf** [comment]
Done.

File: llvm/test/CodeGen/X86/apx/domain-reassignment.mir [hunk 1]
```diff
@@ -524,14 +524,14 @@ body:             |
...
-    %9 = OR32rr %8, %6, implicit-def dead $eflags
-    %10 = AND32rr %9, %7, implicit-def dead $eflags
-    %11 = XOR32rr %10, %6, implicit-def dead $eflags
+    %6 = SHR32ri_ND %5, 2, implicit-def dead $eflags
+    %7 = SHL32ri_ND %6, 1, implicit-def dead $eflags
+    %8 = NOT32r_ND %7
+    %9 = OR32rr_ND %8, %6, implicit-def dead $eflags
+    %10 = AND32rr_ND %9, %7, implicit-def dead $eflags
+    %11 = XOR32rr_ND %10, %6, implicit-def dead $eflags
     %12 = ANDN32rr %11, %9, implicit-def dead $eflags
```
Comment (KanRobert) [inline llvm/test/CodeGen/X86/apx/domain-reassignment.mir:533]: Why this is not changed?

Comment (FreddyLeaf) [reply llvm/test/CodeGen/X86/apx/domain-reassignment.mir:533]: No NDD version for ANDN?

**Review by KanRobert — APPROVED** [review]
LGTM

**Pull request merged**